### PR TITLE
Log failed commands

### PR DESCRIPTION
**Pull request by AndrewP-GH**
Add exception logging with elapsed time for commands execution

### CODE REVIEW

**Comment by aurokk** [comment]
@hryz

File: src/Dapper.Logging/LoggingHook.cs [hunk 1]
```diff
@@ -19,26 +19,37 @@ public LoggingHook(ILogger logger, DbLoggingConfiguration config)
...
+                _config.ExecuteQueryMessage,
+                command.CommandText,
+                command.GetParameters(hideValues: !_config.LogSensitiveData),
+                elapsedMs,
+                context,
+                _connectionProjector(command.Connection));
+
+        public void CommandFailed(DbCommand command, T context, object elapsedMs, Exception exception) =>
+            _logger.Log(
+                _config.LogLevel,
```
Comment (hryz) [inline src/Dapper.Logging/LoggingHook.cs:50]: Should it be the `error` level? Writing errors with the same severity as the rest of the messages might be confusing.

File: src/Dapper.Logging/LoggingHook.cs [hunk 2]
```diff
@@ -19,26 +19,37 @@ public LoggingHook(ILogger logger, DbLoggingConfiguration config)
...
+                exception,
+                _config.ExecuteQueryMessage,
```
Comment (hryz) [inline src/Dapper.Logging/LoggingHook.cs:52]: We might also need a separate message template for errors.

File: src/Dapper.Logging/Hooks/ISqlHooks.cs [hunk 3]
```diff
@@ -8,5 +8,6 @@ public interface ISqlHooks<in T>
         void ConnectionOpened(DbConnection connection, T context, long elapsedMs);
         void ConnectionClosed(DbConnection connection, T context, long elapsedMs);
         void CommandExecuted(DbCommand command, T context, long elapsedMs);
+        void CommandFailed(DbCommand command, T context, object elapsedMs, Exception exception);
```
Comment (hryz) [inline src/Dapper.Logging/Hooks/ISqlHooks.cs:11]: Adding a method to an interface is a breaking change.
Should we have an empty body here using the _default interface members_ language feature? Also, using an `object` for elapsedMs seems like an error.

**Review by hryz — CHANGES_REQUESTED** [review]
Thank you for a nice Idea! I'll think about the best design regarding my comments and get back to this PR.

File: src/Dapper.Logging/Hooks/ISqlHooks.cs (hunk 3 above)
Comment (hryz) [reply src/Dapper.Logging/Hooks/ISqlHooks.cs:11]: I've completely forgotten that the default interface members require `net standard 2.1` and increasing it is also a breaking change.

**Comment by hryz** [comment]
@aurokk do you need this functionality?
I see that @AndrewP-GH gave up and forked this repo 20 days ago 😁.
I can spend a few hours reimplementing this functionality in a way that won't break anyone if you need it.

**Comment by aurokk** [comment]
Hello, @hryz!
Yeah, it's very useful to see logs for failed requests, so it would be nice if you add this functionality :)

**Comment by hryz** [comment]
OK, I'll do it in a few days (I hope 😉)

**Comment by hryz** [comment]
The same effect as proposed here is achieved in #12 but without breaking changs.
The updated version is here https://www.nuget.org/packages/Dapper.Logging/0.4.3

**Comment by hryz** [comment]
@aurokk check out the updated package https://www.nuget.org/packages/Dapper.Logging/0.4.3

**Comment by aurokk** [comment]
@hryz Thank you, works great! 🔥

**Pull request closed without merging**